docs/plus/pv.md

```diff
@@ -0,0 +1,76 @@
+# PersistentVolume
+
+A `PersistentVolume` (PV) is a piece of storage in the cluster that has been provisioned by an administrator or dynamically provisioned using Storage Classes.
+
+!!! tip ""
+    [API Reference](../reference/cdk8s-plus-22/typescript.md#persistent-volume)
+
+The `PersistentVolume` construct represents a pre-existing volume in the cluster.
+
+## Types
+
+Each type is implmented as its own construct, exposing both common properties as well as type
+specific ones. Currently the supported types are:
+
+- `AwsElasticBlockStorePersistentVolume`
+- `AzureDiskPersistentVolume`
+- `GCEPersistentDiskPersistentVolume`
+
+For example, to create a PV from an existing AWS EBS volume:
+
+```ts
+import * as kplus from 'cdk8s-plus-22';
+import * as cdk8s from 'cdk8s';
+
+const vol = new kplus.AwsElasticBlockStorePersistentVolume(chart, 'Volume', {
+  // must exist in aws
+  volumeId: 'vol1234',
+
+  // assign the volume to small-ebs storage class
+  storageClassName: 'small-ebs',
+
+  // what is the volume storage
+  storage: cdk8s.Size.gibibytes(50),
+});
+```
+
+Note that this **does not** actually create a new volume, it merely manifests an existing
+volume in AWS as a Kubernetes resource.
+
+## Reserve
+
+> See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume
+
+Once the PV is defined, you can reserve it:
+
+```ts
+const claim = vol.reserve();
+```
+
+This method creates a new `PersistentVolumeClaim` and performs a
+bi-directional binding that reserves the volume for usage.
+You can use the claim to mount a volume onto a container like usual:
+
+```ts
+container.mount('/data', kplus.Volume.fromPersistentVolumeClaim(claim));
+```
+
+You can also directly mount a persistent volume, which will implicitly reserve it:
+
+```ts
+container.mount('/data', kplus.Volume.fromPersistentVolume(vol));
+```
```

Comment:
I agree its a bit confusing. Added a note that hopefully clears it up a bit. (https://stackoverflow.com/a/5450075 points to some UI question, guess you meant a different link?).

> I wonder if we should just be able to pass vol directly (since "Volume.fromXxxVolume" sounds redundant at first glance)?

Good idea, it required some rewiring but I think I came up with something reasonable. Let me know what you think.
Unfortunately the original PR was already merged by the time I got to it, but here is a new one: https://github.com/cdk8s-team/cdk8s-plus/pull/491

Reply:
Whoops, the URL should have been https://stackoverflow.com/a/54500752 ("What is the difference between a volume and persistent volume?")

The updated API still looks good, though I'm also wondering if it will be confusing that you can do:

```ts
container.mount('/data', pv);
```

which implicitly creates a PVC for you and mounts it -- but if you have a PVC already you need to do:

```ts
container.mount('/data', Volume.fromPersistentVolumeClaim(pvc))
```

What do you think of either:
- Make PVC implement `IStorage` (I don't think this mental model makes sense)
- Replace`IStorage` with `IMountable` and make `mount()` accept any `IMountable` (which could be either a Volume or PVC -- or possibly PV).
- Keep the existing API which makes the volume sources always explicit (drop this PR)

Reply:
As discussed offline, we decided its ok to have `mount` accept a PV, but not a PVC. The mental model of users is mounting a volume, not a claim, which is somewhat of an implementation details for this API. 

We'll keep this API for now and reconsider at a later point if needed.